src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionListProvider.cs

```diff
@@ -94,13 +107,36 @@ internal class RazorCompletionListProvider
         return completionList;
     }
 
+    private async Task<ImmutableArray<CompletionItem>> GetSnippetCompletionsAsync(TextDocumentIdentifierAndVersion identifier, RazorLanguageKind languageKind, SyntaxNode? node, CancellationToken cancellationToken)
+    {
+        if (node is null)
+        {
+            return ImmutableArray<CompletionItem>.Empty;
+        }
+
+        var text = node.ToFullString().Trim();
+
+        var snippetParams = new RazorSnippetCompletionParams(identifier.TextDocumentIdentifier, languageKind, node.Span);
+        var completions = await _clientNotifierService.SendRequestAsync<RazorSnippetCompletionParams, CompletionItem[]?>(
```

Comment:
The main reason I added a new endpoint is to keep our telemetry more accurate. We track completion for dotnet/html as a sub section of our requests. We definitely could change the telemetry for that to be on the client side to report numbers though. 

For VS Code snippets are supported natively by the [client](https://code.visualstudio.com/docs/editor/userdefinedsnippets). We only need to do this work for VS support. Could definitely have a check for if this should be called for completion or not and put false for VS Code to reduce the work.

Reply:
I don't think telemetry is effected here. We track the entire Razor request for completion, but we'd want that to include snippets no matter which side they come from. We then track sub-requests from the custom message target to Roslyn/WebTools, but the snippet endpoint doesn't do any of that, so I think it would be unaffected. 

To put it more concretely, we track sub-request telemetry on line 127 of `Completion.cs`, so as long as we call the new `GetSnippetCompletions` method before that, I think it should be fine.